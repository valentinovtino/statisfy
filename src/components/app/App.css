.App {
  text-align: center;
}
.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}
.App-title {
  font-family: 'title';
  font-size: 10em;
  color: #324E46;
  text-decoration: none;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

@font-face {
  font-family: title;
  src: url(../../Fonts/PWYummyDonuts2.ttf);
}
/*  */



.book-title {
  font-family: 'Boogaloo', cursive;
  font-size: xx-large;
}

.navs{
  color: #324E46;
  display: flex;
  margin-left: 50px;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}


.fave{
  color: #324E46;
  display: flex;
  justify-content: flex-end;

}
.fave-btn {
  /* display: flex; */
  align-self: flex-start;
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

.slogan{
  font-family: 'Boogaloo', cursive;
  font-size: 2em;
  margin-top: 0;
  margin-left: 200px;
  
}

.food-btn {
  border: none;
  font-family: 'Boogaloo', cursive;
  font-size: 2.5em;
  margin-right: 30px;
  background-color: transparent;
  text-decoration: none;
}

.active, .food-btn:hover {
  color: #324E46;
}


