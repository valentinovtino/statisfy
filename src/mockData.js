export const mockObj = {
  "count": 30,
  "recipes": [
    {
      "publisher": "Two Peas and Their Pod",
      "f2f_url": "http://food2fork.com/view/54373",
      "title": "Focaccia Bread",
      "source_url": "http://www.twopeasandtheirpod.com/focaccia-bread/",
      "recipe_id": "54373",
      "image_url": "http://static.food2fork.com/FocacciaBread36db7.jpg",
      "social_rank": 99.99172380177085,
      "publisher_url": "http://www.twopeasandtheirpod.com"
    }
  ]
};

export const mockCleanedDataObj = 
    [{
      "publisher": "All Recipes",
      "f2f_url": "http://food2fork.com/view/12142",
      "title": "Easy Indian Butter Chicken",
      "source_url": "http://allrecipes.com/Recipe/Easy-Indian-Butter-Chicken/Detail.aspx",
      "recipe_id": "12142",
      "image_url": "http://static.food2fork.com/40062066fa.jpg",
      "social_rank": 99.99792034435711,
      "publisher_url": "http://allrecipes.com"
    }];

export const userCleanedData =
[{"id": "54373",
  "img": "http://static.food2fork.com/FocacciaBread36db7.jpg", 
  "recipe": "http://www.twopeasandtheirpod.com/focaccia-bread/", 
  "title": 
 "Focaccia Bread"}];
