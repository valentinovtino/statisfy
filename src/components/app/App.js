import React, { Component } from 'react';
import './App.css';
import { getFoodData } from '../../apiCalls/api';
import PropTypes from 'prop-types';
import FoodHolder from '../../containers/foodHolderContainer';

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      test: 'testing'
    }
  }

  makeFetchSweet = async () => {
    const url = `https://food2fork.com/api/search?key=e3e6267f800bf94ed1db6763e826e406&q=sweet`;
    const response = await getFoodData(url);
    const data = response;


    this.props.storeFood(data.recipes);
  }

  makeFetchSavory = async () => {
    const url = 'https://food2fork.com/api/search?key=e3e6267f800bf94ed1db6763e826e406&q=savory'
    const response = await getFoodData(url);
    const data = response;


    this.props.storeFood(data.recipes);
  }

  makeFetchNutty = async () => {
    const url = 'https://food2fork.com/api/search?key=e3e6267f800bf94ed1db6763e826e406&q=nutty'
    const response = await getFoodData(url);
    const data = response;


    this.props.storeFood(data.recipes);
  }

  render() {
    const { food } = this.props

   
    
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">SATISY</h1>
        </header>

        
        <div className="book">
          <span className="page turn"></span>
          <span className="page turn"></span>
          <span className="page turn"></span>
          <span className="page turn">~~~~~</span>
          <span className="page turn"></span>
          <span className="page turn">
            <h3>What're you craving?..</h3>
            <button onClick={this.makeFetchSweet}>SWEET</button>
            <button onClick={this.makeFetchSavory}>SAVORY</button>
            <button onClick={this.makeFetchNutty}>NUTTY</button>
          </span>
          <span className="cover"></span>
          <span className="page"></span>
          <span className="cover turn"></span>
        </div>

        <div>
          { food.length >0 ?
            <FoodHolder allState={this.state}/> :
            <div>
            </div>
          }
        </div>
       
      </div>
    );
  }
}

export default App;
