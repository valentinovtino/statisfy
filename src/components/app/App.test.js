import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../containers/appContainer';
import { shallow, mount } from 'enzyme';
import { mockObj } from '../../mockData.js';


describe('App', () => {
  let wrapper;
  let mockStoreFood = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App storeFood={mockStoreFood}/>, {disableLifecycleMethods: true});
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Savory button', () => {
    wrapper = shallow(<App storeFood={mockStoreFood}/>, {disableLifecycleMethods: true});
    expect(wrapper.find('.savory-btn').length).toEqual(1)
  });

  it('should storeFood when called', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({json: () => Promise.resolve(mockObj)}))

    await wrapper.instance().makeFetchSweet();

    expect(mockStoreFood).toHaveBeenCalledWith(mockObj.recipes);
  });

  it('should render food category btn if there is no value in store', () => {

  });
});