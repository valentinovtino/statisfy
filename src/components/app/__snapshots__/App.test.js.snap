// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    storeFood={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    SATISY
          </h1>
</header>,
        <button
          onClick={[Function]}
>
          SWEET
</button>,
        <button>
          SAVORY
</button>,
        <button>
          NUTTY
</button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            SATISY
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SATISY",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "SATISY",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWEET",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "SWEET",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SAVORY",
        },
        "ref": null,
        "rendered": "SAVORY",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NUTTY",
        },
        "ref": null,
        "rendered": "NUTTY",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        SATISY
            </h1>
</header>,
          <button
            onClick={[Function]}
>
            SWEET
</button>,
          <button>
            SAVORY
</button>,
          <button>
            NUTTY
</button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              SATISY
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SATISY",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "SATISY",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWEET",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "SWEET",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SAVORY",
          },
          "ref": null,
          "rendered": "SAVORY",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NUTTY",
          },
          "ref": null,
          "rendered": "NUTTY",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
